contracts/0.8.9/lib/ReportExitLimitUtils.sol

```diff
@@ -0,0 +1,155 @@
+// 256______________160____________________________128_______________32____________________________ 0
+// |_________________|______________________________|_________________|_____________________________|
+// | maxExitRequests | maxExitRequestsGrowthBlocks | prevExitRequests | prevExitRequestsBlockNumber  |
+// |<--- 96 bits --->|<---------- 32 bits -------->|<--- 96 bits ---->|<----- 32 bits ------------->|
+//
+
+struct ExitRequestLimitData {
+    uint32 prevExitRequestsBlockNumber;       // block number of the previous exit requests
+    uint96 prevExitRequestsLimit;             // limit value (<= `maxExitRequestLimit`) obtained on the previous exit request
+    uint32 maxExitRequestsLimitGrowthBlocks;  // limit regeneration speed expressed in blocks
+    uint96 maxExitRequestsLimit;              // maximum limit value
+}
+
+
+library ReportExitLimitUtilsStorage {
+    using UnstructuredStorage for bytes32;
+
+    uint256 internal constant MAX_EXIT_REQUESTS_LIMIT_OFFSET = 160;
+    uint256 internal constant MAX_EXIT_REQUESTS_LIMIT_GROWTH_BLOCKS_OFFSET = 128;
+    uint256 internal constant PREV_EXIT_REQUESTS_LIMIT_OFFSET = 32;
+    uint256 internal constant PREV_EXIT_REQUESTS_BLOCK_NUMBER_OFFSET = 0;
+
+    function getStorageExitRequestLimit(bytes32 _position)
+        internal
+        view
+        returns (ExitRequestLimitData memory data)
+    {
+        uint256 slotValue = _position.getStorageUint256();
+
+        data.prevExitRequestsBlockNumber = uint32(slotValue >> PREV_EXIT_REQUESTS_BLOCK_NUMBER_OFFSET);
+        data.prevExitRequestsLimit = uint96(slotValue >> PREV_EXIT_REQUESTS_LIMIT_OFFSET);
+        data.maxExitRequestsLimitGrowthBlocks = uint32(slotValue >> MAX_EXIT_REQUESTS_LIMIT_GROWTH_BLOCKS_OFFSET);
+        data.maxExitRequestsLimit = uint96(slotValue >> MAX_EXIT_REQUESTS_LIMIT_OFFSET);
+    }
+
+    function setStorageExitRequestLimit(bytes32 _position, ExitRequestLimitData memory _data) internal {
+        _position.setStorageUint256(
+            (uint256(_data.prevExitRequestsBlockNumber) << PREV_EXIT_REQUESTS_BLOCK_NUMBER_OFFSET) |
+            (uint256(_data.prevExitRequestsLimit) << PREV_EXIT_REQUESTS_LIMIT_OFFSET) |
+            (uint256(_data.maxExitRequestsLimitGrowthBlocks) << MAX_EXIT_REQUESTS_LIMIT_GROWTH_BLOCKS_OFFSET) |
+            (uint256(_data.maxExitRequestsLimit) << MAX_EXIT_REQUESTS_LIMIT_OFFSET)
+        );
+    }
+}
+
+library ReportExitLimitUtils {
+    /**
+    * @notice Calculate exit requests limit
+    * @dev using `_constGasMin` to make gas consumption independent of the current block number
+    */
+    function calculateCurrentExitRequestLimit(ExitRequestLimitData memory _data) internal view returns(uint256 limit) {
+        uint256 exitRequestLimitIncPerBlock;
+        if (_data.maxExitRequestsLimitGrowthBlocks != 0) {
+            exitRequestLimitIncPerBlock = _data.maxExitRequestsLimit / _data.maxExitRequestsLimitGrowthBlocks;
```

Comment:
wouldn't it be easier to store just the value of `exitRequestLimitIncPerBlock`?
since in `setExitReportLimit` we first recalculate to `maxExitRequestsLimitGrowthBlocks`, and then back here